test/claim-issuers/claim-to.test.ts

```diff
@@ -0,0 +1,166 @@
+      const { claimIssuer, aliceWallet, aliceClaim666, aliceIdentity } =
+        await loadFixture(deployIdentityFixture);
+
+      const aliceKey = ethers.keccak256(
+        ethers.AbiCoder.defaultAbiCoder().encode(
+          ["address"],
+          [aliceWallet.address],
+        ),
+      );
+
+      await identity.addKey(aliceKey, 3, 1); // 3 is CLAIM purpose, 1 is ECDSA type
+      await identity.addKey(aliceKey, 1, 1); // 1 is MANAGEMENT purpose, 1 is ECDSA type
+
+      const claimIssuerKey = ethers.keccak256(
+        ethers.AbiCoder.defaultAbiCoder().encode(
+          ["address"],
+          [await claimIssuer.getAddress()],
+        ),
+      );
+      await identity.addKey(claimIssuerKey, 1, 1);
+      await identity.addKey(claimIssuerKey, 3, 1);
+
+      await aliceIdentity.connect(aliceWallet).addKey(claimIssuerKey, 1, 1);
+      await aliceIdentity.connect(aliceWallet).addKey(claimIssuerKey, 3, 1);
+      // Get the transaction
+      const tx = await claimIssuer.addClaimTo(
+        aliceClaim666.topic,
+        aliceClaim666.scheme,
+        aliceClaim666.issuer,
+        aliceClaim666.signature,
+        aliceClaim666.data,
+        aliceClaim666.uri,
+        aliceClaim666.identity,
+      );
+
+      // Wait for the transaction to be mined and get the receipt
+      const receipt = await tx.wait();
+      // Find the ClaimAdded event
+      const claimAddedEvent = receipt?.logs.find(
+        (log: any) => log.fragment?.name === "ClaimChanged",
+      );
+
+      // Get the claimId from the event using proper type casting
+      const claimId = (claimAddedEvent as any)?.args?.[0];
+
+      const claim = await aliceIdentity.getClaim(claimId);
+      expect(claim.topic).to.equal(aliceClaim666.topic);
+      expect(claim.scheme).to.equal(aliceClaim666.scheme);
+      expect(claim.issuer).to.equal(aliceClaim666.issuer);
+      expect(claim.signature).to.equal(aliceClaim666.signature);
+      expect(claim.data).to.equal(aliceClaim666.data);
+      expect(claim.uri).to.equal(aliceClaim666.uri);
+    });
+
+    it("should revert if not called by manager", async function () {
+      const { claimIssuer, aliceClaim666 } = await loadFixture(
+        deployIdentityFixture,
+      );
+      const otherKey = ethers.keccak256(
+        ethers.AbiCoder.defaultAbiCoder().encode(["address"], [other.address]),
+      );
+      await claimIssuer.removeKey(otherKey, 1);
+
+      const OtherWallet = await ethers.getSigner(other.address);
+
+      await expect(
+        claimIssuer
+          .connect(OtherWallet)
+          .addClaimTo(
+            aliceClaim666.topic,
+            aliceClaim666.scheme,
+            aliceClaim666.issuer,
+            aliceClaim666.signature,
+            aliceClaim666.data,
+            aliceClaim666.uri,
+            await identity.getAddress(),
+          ),
+      ).to.be.revertedWithCustomError(
+        claimIssuer,
+        "SenderDoesNotHaveManagementKey",
+      );
```

Comment:
there is a very important test missing : `addClaimTo` is called by the right key on ClaimIssuer side (so it is executed) but the ClaimIssuer doesn't have the required key on the Identity side to trigger the auto `approve` => should keep the execution of the `addClaim` pending and the Identity owner should be able to call `approve` on it to finalize the claim addition.